<launch>

  <arg name="show_debug_info" default="false"/>
  <arg name="enable_yaw_bias_estimation" default="True"/>
  <arg name="predict_frequency" default="50.0"/>
  <arg name="tf_rate" default="10.0"/>
  <arg name="extend_state_step" default="50"/>

  <!-- for Pose measurement -->
  <arg name="use_pose_with_covariance" default="false"/>
  <arg name="pose_additional_delay" default="0.0"/>
  <arg name="pose_measure_uncertainty_time" default="0.01"/>
  <arg name="pose_rate" default="10.0"/>
  <arg name="pose_gate_dist" default="10000.0"/>
  <arg name="pose_stddev_x" default="0.05"/>
  <arg name="pose_stddev_y" default="0.05"/>
  <arg name="pose_stddev_yaw" default="0.025"/>

  <!-- for twist measurement -->
  <arg name="twist_additional_delay" default="0.0"/>
  <arg name="twist_rate" default="10.0"/>
  <arg name="twist_gate_dist" default="10000.0"/>
  <arg name="twist_stddev_vx" default="0.2"/>
  <arg name="twist_stddev_wz" default="0.03"/>

  <!-- for process model -->
  <arg name="proc_stddev_yaw_c" default="0.005"/>
  <arg name="proc_stddev_yaw_bias_c" default="0.001"/>
  <arg name="proc_stddev_vx_c" default="0.1"/>
  <arg name="proc_stddev_wz_c" default="0.05"/>


  <node pkg="ekf_localizer" type="ekf_localizer" name="ekf_localizer" output="screen">

    <param name="pose_frame_id" value="/map"/>
    <param name="input_pose_name" value="/ndt_pose"/>
    <param name="input_twist_name" value="/can_twist"/>
    <param name="output_pose_name" value="/ekf_pose"/>
    <param name="output_pose_with_covariance_name" value="/ekf_pose_with_covariance"/>
    <param name="output_twist_name" value="/ekf_twist"/>

    <param name="show_debug_info" value="$(arg show_debug_info)"/>
    <param name="enable_yaw_bias_estimation" value="$(arg enable_yaw_bias_estimation)"/>

    <param name="predict_frequency" value="$(arg predict_frequency)"/>
    <param name="tf_rate" value="$(arg tf_rate)"/>
    <param name="extend_state_step" value="$(arg extend_state_step)"/>

    <param name="use_pose_with_covariance" value="$(arg use_pose_with_covariance)"/>
    <param name="pose_additional_delay" value="$(arg pose_additional_delay)"/>
    <param name="pose_measure_uncertainty_time" value="$(arg pose_measure_uncertainty_time)"/>
    <param name="pose_rate" value="$(arg pose_rate)"/>
    <param name="pose_gate_dist" value="$(arg pose_gate_dist)"/>
    <param name="pose_stddev_x" value="$(arg pose_stddev_x)"/>
    <param name="pose_stddev_y" value="$(arg pose_stddev_y)"/>
    <param name="pose_stddev_yaw" value="$(arg pose_stddev_yaw)"/>

    <param name="twist_additional_delay" value="$(arg twist_additional_delay)"/>
    <param name="twist_rate" value="$(arg twist_rate)"/>
    <param name="twist_gate_dist" value="$(arg twist_gate_dist)"/>
    <param name="twist_stddev_vx" value="$(arg twist_stddev_vx)"/>
    <param name="twist_stddev_wz" value="$(arg twist_stddev_wz)"/>

    <param name="proc_stddev_yaw_c" value="$(arg proc_stddev_yaw_c)"/>
    <param name="proc_stddev_yaw_bias_c" value="$(arg proc_stddev_yaw_bias_c)"/>
    <param name="proc_stddev_vx_c" value="$(arg proc_stddev_vx_c)"/>
    <param name="proc_stddev_wz_c" value="$(arg proc_stddev_wz_c)"/>

  </node>

</launch>